src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -201,7 +212,7 @@ private void schedule() {
                     connection.ensureWindowUpdated(df); // must update connection window
                     Log.logTrace("responseSubscriber.onComplete");
                     if (debug.on()) debug.log("incoming: onComplete");
-                    sched.stop();
+                    if (inputQ.isEmpty()) sched.stop();
```

Comment:
I think you should remove all references to sched.stop() from this method; while adding `if (inputQ.isEmpty())` reduces the risk of races, there's still a small window where an incoming reset frame can be lost.

Reply:
The only thing we could with the reset after this point in the schedule() method, if the reset had not yet been inserted into the inputQueue, is to complete the `requestBodyCF` and this would have already been taken care of at line 595 in incoming_reset. So I don't believe the scheduler would need to run again after this line (or the other like it)

Reply:
I think maybe its better to let the scheduler complete without these checks. I think the effect of `stop()` is that it makes all subsequent calls to `runOrSchedule()` no-ops. I think it might be best to remove the call to stop() and allow scheduler to complete the requestBodyCf in a similar mannet to how its done in `incoming_reset()`.